metricbeat/tests/system/test_haproxy.py

```diff
@@ -87,7 +87,15 @@ def test_stat_http_auth(self):
             "metricsets": ["stat"],
             "username": "admin",
             "password": "admin",
-            "hosts": ["http://%s:%d/stats" % (os.getenv('HAPROXY_HOST', 'localhost'), 14569)],
+            "hosts": ["http://%s:%d/stats" % (self.compose_hosts()[0], 14569)],
             "period": "5s"
         }])
         self._test_stat()
+
+
+class Haproxy_1_6_Test(HaproxyTest):
```

Comment:
Not for now, but what if we could autogenerate this part? Meaning we would separate the test code from the versions against we want to test. This is getting interesting :-)

Reply:
There are ways to generate classes, I could give a try, but I'd wait to see more cases, for example in apache we had to override some parts of the test because each version provides different metrics, it was not only changing the scenario.

Reply:
+1 on waiting for some more cases to see how we can generalise it.